analyzers/src/SonarAnalyzer.CSharp/Rules/DeclareEventHandlersCorrectly.cs

```diff
@@ -35,51 +35,49 @@ public sealed class DeclareEventHandlersCorrectly : SonarDiagnosticAnalyzer
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
         protected override void Initialize(SonarAnalysisContext context)
         {
             context.RegisterSyntaxNodeActionInNonGenerated(
-               c => AnalyzeEventType(c, ((EventFieldDeclarationSyntax)c.Node).Declaration.Type),
+               c => AnalyzeEventType(c, ((EventFieldDeclarationSyntax)c.Node).Declaration.Type, c.ContainingSymbol),
                SyntaxKind.EventFieldDeclaration);
 
             context.RegisterSyntaxNodeActionInNonGenerated(
-               c => AnalyzeEventType(c, ((EventDeclarationSyntax)c.Node).Type),
+               c => AnalyzeEventType(c, ((EventDeclarationSyntax)c.Node).Type, c.ContainingSymbol),
                SyntaxKind.EventDeclaration);
         }
 
-        private void AnalyzeEventType(SyntaxNodeAnalysisContext analysisContext, TypeSyntax typeSyntax)
+        private static void AnalyzeEventType(SyntaxNodeAnalysisContext analysisContext, TypeSyntax typeSyntax, ISymbol eventSymbol)
         {
-            var eventHandlerType = analysisContext.SemanticModel.GetSymbolInfo(typeSyntax).Symbol
-                        as INamedTypeSymbol;
-            var methodSymbol = eventHandlerType?.DelegateInvokeMethod;
-            if (methodSymbol == null)
+            if (eventSymbol.IsOverride || eventSymbol.GetInterfaceMember() != null)
             {
                 return;
             }
 
-            if (!IsCorrectEventHandlerSignature(methodSymbol))
+            var eventHandlerType = analysisContext.SemanticModel.GetSymbolInfo(typeSyntax).Symbol as INamedTypeSymbol;
+            if (eventHandlerType?.DelegateInvokeMethod is { } methodSymbol
+                && !IsCorrectEventHandlerSignature(methodSymbol))
             {
-                analysisContext.ReportIssue(Diagnostic.Create(rule, typeSyntax.GetLocation()));
+                analysisContext.ReportIssue(Diagnostic.Create(Rule, typeSyntax.GetLocation()));
             }
         }
 
-        private bool IsCorrectEventHandlerSignature(IMethodSymbol methodSymbol)
-        {
-            return methodSymbol.ReturnsVoid &&
-                methodSymbol.Parameters.Length == 2 &&
-                methodSymbol.Parameters[0].Name == "sender" &&
-                methodSymbol.Parameters[0].Type.Is(KnownType.System_Object) &&
-                methodSymbol.Parameters[1].Name == "e" &&
-                IsDerivedFromEventArgs(methodSymbol.Parameters[1].Type);
-        }
+        private static bool IsCorrectEventHandlerSignature(IMethodSymbol methodSymbol) =>
+            methodSymbol.ReturnsVoid
+            && methodSymbol.Parameters.Length == EventArgumentPosition
```

Comment:
`0` and `1` are magic numbers here too. Replacing `2` only with a constant, do not make it easier to read.